### PR TITLE
fix/clean-up-bun-error

### PR DESCRIPTION
**Pull request by JohnDaly**
Updates to the `bun-error` package to remove unused code and fix TypeScript errors.

There are quite a few changes, but this is the general breakdown:
1. Removing unused imports
2. Adding missing prop types to components
3. Removing unused props from components
4. Fixing bugs caught by TypeScript

### CODE REVIEW

File: packages/bun-error/index.tsx [hunk 1]
```diff
@@ -1137,7 +1088,7 @@ const OverlayMessageContainer = ({
 function copyToClipboard(input: string | Promise<string>) {
   if (!input) return;
 
-  if (input && typeof input === "object" && "then" in input) {
+  if (typeof input === "object" && "then" in input) {
```
Comment (Jarred-Sumner) [inline packages/bun-error/index.tsx:1088]: If `input` was null, this code would throw at runtime so the `input &&` is necessary

File: packages/bun-error/index.tsx [hunk 2]
```diff
@@ -827,7 +785,6 @@ const NativeStackTrace = ({
           highlightColumnStart={position.column_start}
           buildURL={buildURL}
           highlightColumnEnd={position.column_stop}
-          isClient={isClient}
```
Comment (Jarred-Sumner) [inline packages/bun-error/index.tsx:830]: There was a reason for this, but I don't remember it anymore

File: packages/bun-error/index.tsx [hunk 3]
```diff
@@ -257,7 +255,7 @@ class FancyTypeError {
   constructor(exception: JSException) {
     this.runtimeType = exception.runtime_type || 0;
     this.runtimeTypeName = RuntimeType[this.runtimeType] || "undefined";
-    this.message = exception.message;
+    this.message = exception.message ?? '';
```
Comment (Jarred-Sumner) [inline packages/bun-error/index.tsx]:
```suggestion
    this.message = exception.message || '';
```

File: packages/bun-error/index.tsx [hunk 4]
```diff
@@ -1439,7 +1390,12 @@ export function dismissError() {
       runtimeErrorController = null;
     }
 
-    while (pending.length > 0) pending.shift().stopThis = true;
+    while (pending.length > 0) {
+      const pendingFrame = pending.shift();
+      if (pendingFrame) {
```
Comment (Jarred-Sumner) [inline packages/bun-error/index.tsx]: this branch is unnecessary because we already check `pending.length > 0`

**Comment by Jarred-Sumner** [comment]
thank you for this

File: packages/bun-error/index.tsx (hunk 1 above)
Comment (JohnDaly) [reply packages/bun-error/index.tsx:1088]: The `if (!input) return` guard on line 1086 should cover that case, so the additional check shouldn't be necessary

File: packages/bun-error/index.tsx (hunk 4 above)
Comment (JohnDaly) [reply packages/bun-error/index.tsx]: Fixed in https://github.com/oven-sh/bun/pull/753/commits/f4c3f73806e63f15ddb88d6893afbdfb2381a4b1

File: packages/bun-error/index.tsx (hunk 3 above)
Comment (JohnDaly) [reply packages/bun-error/index.tsx]: Fixed in https://github.com/oven-sh/bun/pull/753/commits/f4c3f73806e63f15ddb88d6893afbdfb2381a4b1

File: packages/bun-error/index.tsx (hunk 2 above)
Comment (JohnDaly) [reply packages/bun-error/index.tsx:830]: `SourceLines` doesn't currently use the prop. Should I put this back, in case there are future plans for it?

Comment (Jarred-Sumner) [reply packages/bun-error/index.tsx:830]: no this is fine

**Comment by Jarred-Sumner** [comment]
Thanks

Heads up that there will be a CLA coming soon and I will ping you asking to sign it once ready

**Pull request merged**